Meta/CMake/common_compile_options.cmake

```diff
@@ -5,42 +5,54 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 set(CMAKE_COLOR_DIAGNOSTICS ON)
 
-add_compile_options(-Wall)
-add_compile_options(-Wextra)
+# FIXME: when compiing using cl-clang, the code tries to compile c++98 mode, this generates too much warnings
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang$" AND NOT CMAKE_CXX_SIMULATE_ID MATCHES "MSVC")
+    add_compile_options(-Wall)
+    add_compile_options(-Wextra)
+endif()
```

Comment:
if (MSVC)
  add(/W4)
  else()
  add(-Wall -Wextra)
  endif()
  
  Not sure why we're going to the trouble of checking CMAKE_CXX_SIMULATE_ID?
  
  I'm pretty sure we have semi-working flags on both cammo1123's branch, and in jakt. why not just pull those in?

Reply:
done in current iteration of this PR.